app/controllers/request-to-go-live/organisation-address/post.controller.js

```diff
@@ -0,0 +1,97 @@
+'use strict'
+
+// NPM dependencies
+const lodash = require('lodash')
+
+// Local dependencies
+const { requestToGoLive } = require('../../../paths')
+const {
+  validateMandatoryField, validateOptionalField, validatePostcode, validatePhoneNumber
+} = require('../../../utils/validation/server-side-form-validations')
+const { renderErrorView } = require('../../../utils/response')
+
+const ADDRESS_LINE1_FIELD = 'address-line1'
+const ADDRESS_LINE2_FIELD = 'address-line2'
+const ADDRESS_CITY_FIELD = 'address-city'
+const ADDRESS_COUNTRY_FIELD = 'address-country'
+const ADDRESS_POSTCODE_FIELD = 'address-postcode'
+const TELEPHONE_NUMBER_FIELD = 'telephone-number'
+
+const validationRules = [
+  {
+    field: ADDRESS_LINE1_FIELD,
+    validator: validateMandatoryField,
+    maxLength: 200
+  },
+  {
+    field: ADDRESS_LINE2_FIELD,
+    validator: validateOptionalField,
+    maxLength: 200
+  },
+  {
+    field: ADDRESS_CITY_FIELD,
+    validator: validateMandatoryField,
+    maxLength: 100
+  },
+  {
+    field: ADDRESS_COUNTRY_FIELD,
+    validator: validateMandatoryField,
+    maxLength: 2
+  },
+  {
+    field: TELEPHONE_NUMBER_FIELD,
+    validator: validatePhoneNumber
+  }
+]
+
+const validate = function validate (formFields) {
+  const errors = {}
+  validationRules.forEach(validationRule => {
+    const value = formFields[validationRule.field]
+    const validationResponse = validationRule.validator(value, validationRule.maxLength)
+    if (!validationResponse.valid) {
+      errors[validationRule.field] = validationResponse.message
+    }
+  })
+
+  const postCode = formFields[ADDRESS_POSTCODE_FIELD]
+  const country = formFields[ADDRESS_COUNTRY_FIELD]
+  const postCodeValidResponse = validatePostcode(postCode, country)
+  if (!postCodeValidResponse.valid) {
+    errors[ADDRESS_POSTCODE_FIELD] = postCodeValidResponse.message
+  }
+  return errors
+}
+
+module.exports = (req, res) => {
+  const trimField = (fieldName) => {
```

Comment:
A small performance comment but this could be declared outside of the HTTP response controller so that it's not initialised on every request. 

It could also be slightly simplified, something like:
``` 
const trimField = (key, store) => lodash.get(store, key, '').trim()
```

With the frequency of these requests it's likely negligible so up to you.

Reply:
👍 made this change